src/Components/Tasks/Tasks.tsx

```diff
@@ -1,7 +1,130 @@
-import React from 'react'
+// nadia.mohamed.taha166@gmail.com
+// @Password123!
 
-export default function Tasks() {
+//token
+//eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjYsInJvbGVzIjpbIk1hbmFnZXIiLCJjYW5BZGRVc2VyIiwiY2FuVXBkYXRlVXNlciIsImNhbkRlbGV0ZVVzZXIiLCJjYW5HZXRVc2VyQnlJZCIsImNhbkdldEN1cnJlbnRVc2VyIiwiY2FuR2V0QWxsVXNlcnMiLCJjYW5DaGFuZ2VQYXNzd29yZCJdLCJ1c2VyTmFtZSI6Im5hZGlhMSIsInVzZXJFbWFpbCI6Im5hZGlhLm1vaGFtZWQudGFoYTE2NkBnbWFpbC5jb20iLCJ1c2VyR3JvdXAiOiJNYW5hZ2VyIiwiaWF0IjoxNzAzMDY4MDcyLCJleHAiOjE3MDY2NjgwNzJ9.ECN8Ispoty_gt3zS4q6hPw0yxLUupaEYCas3hbJ_Y8IXapR39ZQi9WA5oyjIrX10IspPUiWa27ne995iQHexj46cyKvmTPE2PdsL7pIR-xt51z2Vt9pbl6UGfjo6Rgj8UAk0_9AFB7aLR8coS4uBDt3iJBvucr7uXXzA334memARCvDRe981NkoKZ5fddkHgEE3OWGFTOjXqxgd8C-mFaxaVtrxaUw_CaXfg4l067__MjDCnbQLO3p88fiYgeQ1OBeLzoJ9V6EuYYqGjTUNiUe_LS2AXqx4dvN6RiFNCYwQX1zqSiU814RziffSuat_ZdL9PJtPuYJukQ0ygb4A-Og
+
+//user//samiya.bouridane@gmail.com
+//@Password123!
+//manager@gmail.com
+//@Password123!
+import axios from "axios";
+import { useContext, useEffect, useState } from "react";
+// import noData from "./../../assets/images/no-data.png";
+// import Modal from "react-bootstrap/Modal";
+import NoData from './../../Shared/noData/noData';
+import { AuthContext } from "./../../Context/AuthContext";
+import { ToastContext } from "../../Context/ToastContext";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+
+
+const Tasks: React.FC  = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const { baseUrl, requestHeaders }: any = useContext(AuthContext);
+  const { getToastValue }: any = useContext(ToastContext);
+  
+  const [tasks, setTasks] = useState([]);
+  const navigate = useNavigate();
+  // const [itemId, setItemId]: any = useState(0);
+
+
+  // const {
+  //   register,
+  //   setValue,
+  //   handleSubmit,
+  //   formState: { errors },
+  // } = useForm<FormValues>();
+
+  // **********get all tasks*****************
+  const getTasksList = () => {
+    setIsLoading(true);
+    axios
+      .get(`${baseUrl}/Task/manager`, 
+      { headers: requestHeaders })
+      .then((response) => {
+        setIsLoading(false);
```

Comment:
You can add the finally keyword after the catch and set the loading state to false in it, instead of writing it in the then and the catch statements